internal/core/command/restDevice.go

```diff
@@ -228,7 +252,15 @@ func restGetCommandsByDeviceName(
 		return
 	}
 
-	w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
+	// Check the header value on the response
+	if r.Header.Get(clients.ContentType) == clients.ContentTypeCBOR {
+		// Content-type CBOR will be sent to the caller.
+		w.Header().Set(clients.ContentType, clients.ContentTypeCBOR)
+	} else if r.Header.Get(clients.ContentType) == clients.ContentTypeJSON {
+		// Content-type JSON will be sent to the caller.
+		w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
+	}
```

Comment:
See my other comment regarding adding `Content-type` response header from the request header.